civicrm.php

```diff
@@ -307,6 +308,13 @@ public function activation() {
      * @since 5.6
      */
     do_action( 'civicrm_activation' );
+    if (get_option('civicrm_activation_in_progress', false)) {
+      delete_option('civicrm_activation_in_progress');
+      if (!isset($_GET['activate-multi'])) {
+        wp_redirect(admin_url("options-general.php?page=civicrm-install"));
+        exit;
+      }
+    }
```

Comment:
@kcristiano I think this block needs to come after:

`update_option( 'civicrm_activation_in_progress', 'false' );`

Otherwise that option will never be updated.

Reply:
Is it not possible just to add:

```php
if (!isset($_GET['activate-multi'])) {
  wp_redirect(admin_url("options-general.php?page=civicrm-install"));
  exit;
}
```

to the end of this method?

Reply:
That does simplfy the code .  I tend to do things the long way.

Reply:
@kcristiano Just wondering if the `civicrm_setup_do_activation_redirect` option is needed at all here? The flow is already handled by the `civicrm_activation_in_progress` option (which intentionally remains in place until deactivation) so I'm wondering what `civicrm_setup_do_activation_redirect` achieves here.

Reply:
The idea was to have it in case the redirect failed to execute.  But I have not had any issues in testing.  Perhaps we should just remove.